### DISCUSSION TITLE
How do I get a "ß"

### DISCUSSION THREAD

**Discussion by mseibert** [evan-liu/karabiner.ts#105] [Q&A] [answered]
I want to turn this line
`duoModifier('ds', '⇧'),`
into this
`duoModifier('ds', 'ß'),`
But it does not work. How do I do it?

**Comment by mseibert** [comment]
Basically, I am searching for an easy way to create ß with Karabiner.ts.

**Reply by evan-liu** [reply, marked as answer]
If you want to create it when you press 'd' and 's' together, you can try 

    mapSimultaneous(['d', 's']).to('s', '⌥'), // If your keyboard input has 'ß', change '⌥s' to what's on your keyboard
    mapSimultaneous(['d', 's']).toPaste('ß'), // Otherwise you can paste it via clipboard

**Reply by mseibert** [reply]
I could find a workaround with this code (via ChatGPT):

`const customDsToSs: { type: 'basic', from: any, to: any[] } = {
  type: 'basic',
  from: {
    simultaneous: [
      { key_code: 'd' },
      { key_code: 's' },
    ],
  },
  to: [
    { key_code: 'left_option', modifiers: ['left_option'] },
    { key_code: 's', modifiers: ['left_option'] },
  ],
};
`

**Reply by evan-liu** [reply]
Sorry, I am unfamiliar with this, as I use an English keyboard layout. It looks like you need `⌥, s` instead of`⌥ + s`. Can you try 

    mapSimultaneous(['d', 's']).to('left_option').to('s')
    // or
    mapSimultaneous(['d', 's']).to('‹⌥').to('s')

If `mapSimultaneous(['d', 's']).to('s', '⌥')` doesn't work

**Reply by mseibert** [reply]
Oh, your solution is elegant and does work. I also now have a US keyboard. But as a German I still have to use those Umlauts. 🤷‍♂️

**Comment by mseibert** [comment]
@evan-liu I have realized that I also need big letters. I have created this:

`  mapSimultaneous([';', 'u', "left_shift"]).toPaste('Ü'),
    mapSimultaneous([';', 'o', "left_shift"]).toPaste('Ö'),
    mapSimultaneous([';', 'a', "left_shift"]).toPaste('Ä'),`

Is there a way that I can press "left shift" first, keep it pressed, and then click ";" and "u" or "a" or "o" simultaneously?

**Reply by evan-liu** [reply]
You can try 

```typescript
modifierLayer('‹⇧', ';').manipulators({
  u: toPaste('Ü'),
  o: toPaste('Ö'),
  a: toPaste('Ä'),
}),
```

However, `‹⇧ + ;` will no longer trigger `:` that way. You can try modifiers other than `⇧` if it is a problem. 

    modifierLayer('⌘', ';')...

Or you can try

```typescript
duoLayer(';', 'f').manipulators({
  u: toPaste('Ü'),
  o: toPaste('Ö'),
  a: toPaste('Ä'),
}),
```

So you can press `;` and `f` together (or any two keys) to trigger the layer, keep them pressed (only need to keep one of them pressed), then press `a/o/u`.

**Reply by mseibert** [reply]
Thank you so much. The second option works like a charm!
